src/commcare_cloud/environment/secrets/backends/ansible_vault/main.py

```diff
@@ -0,0 +1,147 @@
+import getpass
+import os
+import sys
+from contextlib import contextmanager
+
+import datadog.api
+import yaml
+from ansible.parsing.vault import AnsibleVaultError
+from ansible_vault import Vault
+from memoized import memoized
+from six.moves import shlex_quote
+
+from commcare_cloud.environment.paths import ANSIBLE_DIR
+from commcare_cloud.environment.secrets.secrets_schema import get_known_secret_specs
+
+
+class AnsibleVaultSecretsBackend(object):
+    def __init__(self, name, vault_file_path, record_to_datadog=False):
+        self.name = name
+        self.vault_file_path = vault_file_path
+        self.record_to_datadog = record_to_datadog
+        self.should_send_vault_loaded_event = True
+
+    def prompt_user_input(self):
```

Comment:
nit: something like `prompt_for_password` might be more obvious

Reply:
The reason I chose something generic here is that depending on the backend you might prompt for a password here (as the ansible-vault backend does) or for no input at all (if you're fetching secrets from a service where authorization is given based on being on an internal network or making the request from a specific machine) or to just make sure the user knows what they're doing "Do you want to access secrets for staging? If so please type 'staging':" or whatever else happens to be necessary for a particular secrets backend. The point of the function is just to give the calling code a way to force any potential user interaction to happen up front, whatever it may be.

Reply:
huh, okay, thanks for the explanation